pkg/controller/mysql/user/reconciler.go

```diff
@@ -115,6 +117,10 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 		return nil, errors.Wrap(err, errGetSecret)
 	}
 
+	if err := tls.LoadConfig(ctx, c.kube, pc.Spec.TLS, pc.Spec.TLSConfig); err != nil {
```

Comment:
Question: can and should this be called in newDB instead? It's something you always have to call before newDB? :thinking:

Reply:
We _could_ merge the two, but I see `newDB` as a non-failing function with external calls, no error result, etc, and I also perceive those two functions as conceptually different. We would need to refactor `newDB` considerably, along with corresponding unit tests, as it would slightly affect its scope/role. I tried to be as little intrusive as possible with my changes, but if we commonly decide that they should be merged together, I will tackle that refactoring.